src/main/kotlin/com/atlassian/performance/tools/report/api/judge/Verdict.kt

```diff
@@ -14,15 +15,39 @@ class Verdict internal constructor(
     val reports: List<JUnitReport>,
     val failedActions: List<ActionType<*>>
 ) {
+    private val logger = LogManager.getLogger(this::class.java)
 
+    @Deprecated(message = "Use Builder instead", replaceWith = ReplaceWith("Verdict.Builder(reports).build()"))
     constructor(
         reports: List<JUnitReport>
     ) : this(
         reports = reports,
         failedActions = emptyList()
     )
 
-    private val logger = LogManager.getLogger(this::class.java)
+    /**
+     * Since 3.12.0
+     */
+    class Builder(reports: List<JUnitReport>) {
```

Comment:
Every mandatory constructor field makes builder more difficult for the consumers. The `reports` can have a default value, so the constructor can be parameterless. I'd say `emptyList()` is a legit default. Especially since `Verdict` constructors also accept empty lists.

Reply:
Actually, I'll open a PR towards this PR to show what I mean

Reply:
Here it is: https://github.com/atlassian/report/pull/73. I hope it shows the power of a builder.
@mgrzaslewicz please review it and merge if it's good :crossed_fingers: 
I'm approving this PR right now, not to block you for the next few days when I'm off.